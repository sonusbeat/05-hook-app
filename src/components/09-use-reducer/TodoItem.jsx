import PropTypes from 'prop-types';

const TodoItem = ({ HandleToggle, HandleDelete, todo, index }) => {
  return (
    <tr>
      <td>
        <p
          className={todo.done ? "description completed" : "description"}
          onClick={() => HandleToggle(todo.id)}
        >
          { `${index + 1}. ${todo.description}` }
        </p>
      </td>
      <td className="text-center">{todo.done ? "Yes" : "No"}</td>
      <td className="text-center">
        <button
          className="btn btn-danger btn-sm"
          onClick={() => HandleDelete(todo.id)}
        >
          Delete
        </button>
      </td>
    </tr>
  );
};

export default TodoItem;

TodoItem.propTypes = {
  HandleToggle: PropTypes.func.isRequired,
  HandleDelete: PropTypes.func.isRequired,
  todo: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired
};