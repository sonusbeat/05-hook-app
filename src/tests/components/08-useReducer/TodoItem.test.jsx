import TodoItem from '../../../components/09-use-reducer/TodoItem';
import { shallow } from 'enzyme';
import demoTodos from '../../fixtures/demotTodos';

describe('Pruebas en el component <TodoItem />', () => {

  const HandleToggle = jest.fn();
  const HandleDelete = jest.fn();

  const wrapper = shallow(
    <TodoItem
      HandleToggle={ HandleToggle }
      HandleDelete={ HandleDelete }
      todo={ demoTodos[0] }
      index={ 0 }
    />
  );

  test('Deberia coincidir con el Snapshot', () => {

    expect( wrapper ).toMatchSnapshot();

  });

  test('Debe de llamar la función toggle', () => {

    wrapper.find(".description").simulate("click");
    expect( HandleToggle ).toHaveBeenCalledWith(demoTodos[0].id);

  });

  test('Debe de llamar la función borrar', () => {

    wrapper.find(".btn-danger").simulate("click");
    expect( HandleDelete ).toHaveBeenCalledWith(demoTodos[0].id);

  });

  test('Debe de mostrar el texto correctamente', () => {

    const paragraph = wrapper.find(".description");
    expect( paragraph.text().trim() ).toBe(`1. ${demoTodos[0].description}`);

  });

  test('Debe de tener la clase complete si el TODO.done === true', () => {

    const todo = demoTodos[0];
    todo.done = true;

    const wrapper = shallow(
      <TodoItem
        HandleToggle={ HandleToggle }
        HandleDelete={ HandleDelete }
        todo={ todo }
        index={ 0 }
      />
    );

    expect(wrapper.find(".description").hasClass("completed")).toBe(true);
  });

});
