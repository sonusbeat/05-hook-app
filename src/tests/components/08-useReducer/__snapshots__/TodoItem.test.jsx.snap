// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el component <TodoItem /> Deberia coincidir con el Snapshot 1`] = `
<tr>
  <td>
    <p
      className="description"
      onClick={[Function]}
    >
      1. Aprender React
    </p>
  </td>
  <td
    className="text-center"
  >
    No
  </td>
  <td
    className="text-center"
  >
    <button
      className="btn btn-danger btn-sm"
      onClick={[Function]}
    >
      Delete
    </button>
  </td>
</tr>
`;
